flux_local/git_repo.py

```diff
@@ -466,10 +474,110 @@ async def visit_kustomization(
 
     kinds = [CLUSTER_KUSTOMIZE_KIND, CONFIG_MAP_KIND, SECRET_KIND]
 
+    def _normalize_patterns(patterns: list[str]) -> list[str]:
```

Comment:
The changes in this section here and below seem extremely complicated for implementing this feature. It seems like it only needs to be like 5 line change. There are now 3 copies of essentially the same code, so I'd like to decline this, unless it can be 10x smaller.